patroni/validator.py

```diff
@@ -365,6 +365,26 @@ def assert_(condition, message="Wrong value"):
     Optional("pg_ctl_timeout"): int,
     Optional("use_pg_rewind"): bool
   },
+  Optional("slots"): {
+    "name": Or(
+      {
+        "type": "physical",
+      },
+      {
+        "type": "logical",
+        "database": str,
+        "plugin": str,
+      }
+    ),
+  },
+  Optional("ignore_slots"): [
+    {
+      Optional("name"): str,
+      Optional("database"): str,
+      Optional("plugin"): str,
+      Optional("type"): str,
+    }
+  ],
```

Comment:
`slots` and `ignore_slots` are stored in DCS (and also always accessed via cluster.config), therefore validator in its current state will not check them.

Reply:
If they are configured in the yml and you run `--validate-config` would it be checked? Should we instead do validation somewhere else?

Reply:
If they are configured in yaml (`postgresql.slots` and `postgresql.ignore_slots`), this [code](https://github.com/zalando/patroni/pull/1742/files#diff-e77f860f36a4c7c541f165a9ded2335eb58ed020bab147af0a2315d912dde812R52) will not pick it up.

Reply:
I believe both of those are top-level parameters not on the `postgresql` hash, right?

Does that mean you can't configure `slots` at startup? Or just that the way we're accessing it needs to change? Or that we need to load the config into the DCS when we start up?

Reply:
Oh, sorry missed it.
Yes, they are sort of "top" level, but in the global configuration, which is stored in the DCS, in the /config key.
When you run `patronictl edit-config` it works with the /config key.

But, in the Paroni config there is `bootstrap.dcs` section. These parameters could be placed there and Patroni will write the content of bootstrap.dcs to the config key when the new cluster is created (initdb or custom bootstrap).

Having some validation for bootstrap.dcs would be good, but it is a different task.

Reply:
OK, we'll remove the validation for now.